### PR TITLE
flesh out cookie jar

### PR DESCRIPTION
**Pull request by e0ff**

### CODE REVIEW

File: src/cookies.zig [hunk 1]
```diff
@@ -89,48 +90,67 @@ test Cookie {
...
 
+    /// Creates a new jar.
     pub fn init(a: std.mem.Allocator) !Jar {
-        var cookies = try a.alloc(Cookie, 8); // 8 aught to be enough for anyone! :D
-        cookies.len = 0;
         return .{
             .alloc = a,
-            .cookies = cookies,
+            .cookies = ArrayListUnmanaged(Cookie){},
             .capacity = 8,
```
Comment (GrayHatter) [inline src/cookies.zig]: lets delete capacity if we're gonna use array list

File: src/cookies.zig [hunk 2]
```diff
@@ -89,48 +90,67 @@ test Cookie {
...
-    /// Dummy function, not implemented
+    /// Removes a cookie from the jar.
     pub fn remove(jar: *Jar, name: []const u8) ?Cookie {
         var found: ?Cookie = null;
-        for (jar.cookies, 0..) |cookie, i| {
+
+        for (jar.cookies.items, 0..) |cookie, i| {
             if (eql(u8, cookie.name, name)) {
+                _ = jar.cookies.swapRemove(i);
                 found = cookie;
```
Comment (GrayHatter) [inline src/cookies.zig]:
```suggestion
        for (0..jar.cookies.items.len) |i| {
            if (eql(u8, cookie.name, name)) {
                found = jar.cookies.swapRemove(i);                
```

Comment (GrayHatter) [reply src/cookies.zig]: this is much better than I would have done :) I forgot about swap remove

File: src/cookies.zig [hunk 3]
```diff
@@ -89,48 +90,67 @@ test Cookie {
...
-                // TODO copy remaining backwards
-                _ = i;
             }
         }
+
         return found;
     }
+
+    /// Reduces size of cookies to it's current length.
```
Comment (GrayHatter) [inline src/cookies.zig:126]: no notes, but why tho?

File: src/cookies.zig [hunk 4]
```diff
@@ -89,48 +90,67 @@ test Cookie {
...
+    pub fn shrink(jar: *Jar) void {
+        jar.cookies.shrinkAndFree(jar.alloc, jar.cookies.len);
+    }
 };
 
 test Jar {
     const a = std.testing.allocator;
     var j = try Jar.init(a);
```
Comment (GrayHatter) [inline src/cookies.zig:134]: defer raze please?

**Review by GrayHatter — CHANGES_REQUESTED** [review]

File: src/cookies.zig (hunk 4 above)
Comment (e0ff) [reply src/cookies.zig:134]: Done

File: src/cookies.zig (hunk 1 above)
Comment (e0ff) [reply src/cookies.zig]: Done

**Pull request merged**